src/codegen/compile.ml

```diff
@@ -12138,8 +11859,8 @@ enabled mutual recursion.
 and compile_lit_pat env l =
   match l with
   | NullLit ->
-    compile_lit_as env SR.Vanilla l ^^
-    G.i (Compare (Wasm.Values.I32 I32Op.Eq))
+    Opt.is_some env ^^
+    Bool.neg
   | BoolLit true ->
```

Comment:
Nit: why not just compare against null value/ptr? I'm not sure the peephole optimizer catches this.

Reply:
Thanks, I simplified this. (Probably a legacy from when I used a dedicated null singleton object.)